term.go

```diff
@@ -354,6 +354,11 @@ func (term *Terminal) prompt(buf *Buffer, in io.Reader) (string, error) {
 			continue
 		case evReturn:
 			err = buf.EndLine()
+			if buf.String() == "" {
+				term.History = term.History[:term.histIdx]
+				term.histIdx = term.histIdx - 1
+				return buf.String(), err
+			}
 			term.History[curHistIdx] = buf.String()
```

Comment:
Does it not work to simply wrap this line in a `if buf.String != ""` check?

Reply:
Roll back is required to handle decrementing idx as it increments it before the switch https://github.com/Bowery/prompt/blob/master/term.go#L335